### ISSUE TITLE
make_artifacts.py: split is not reproducible across environments despite the seed

### ISSUE THREAD

**Issue by GIND123** [GIND123/Inference-Guard#21]
Follow-up from the review of #15. Not blocking Session 04; should land before we report any number measured on the test split.

`scripts/make_artifacts.py` regenerates `artifacts/profile_splits.json` and is seeded, but the seed is not sufficient to reproduce the split.

```python
unique_profiles = list(profile_to_examples.keys())
...
train_p, temp_p = train_test_split(unique_profiles, test_size=0.30, random_state=SEED)
```

`train_test_split` shuffles deterministically, but it shuffles the list it is given, and that list's order is dict insertion order, which is dataset iteration order. A different iteration order with the same seed produces a different 210/45/45. It currently agrees with the committed JSON only because everyone is hitting the same cached dataset revision.

This matters because the held-out test split is what every privacy claim in the project eventually rests on. If the split silently moves, a Session 09 held-out number is not comparable to a Session 05 one, and we would have no way to notice.

### Fix

- [ ] `unique_profiles = sorted(profile_to_examples)` — order no longer depends on the loader
- [ ] Pin the dataset: `load_dataset("RobinSta/SynthPAI", revision="<commit sha>")`
- [ ] Record that revision in `profile_splits.json` next to `seed`, so the artifact states what it was built from
- [ ] Re-run and confirm the regenerated split matches the committed one byte for byte; if it does not, say so in the Session 05 report rather than quietly replacing the file

### Also in this file

- [ ] `random.seed(SEED)` and `np.random.seed(SEED)` are dead code — `train_test_split(random_state=...)` uses its own RNG and never reads the globals. Remove them or they imply a determinism guarantee the script does not provide.
- [ ] The `assert len(unique_profiles) == 300` / `210/45/45` checks fail loud if the dataset changes, which is correct behaviour once the revision is pinned. Worth a one-line comment saying that is deliberate, so nobody "fixes" it by relaxing the assert.

### Related

The design doc's reproducibility requirements (seeds, model versions, dataset hashes, config, experiment IDs) are only partly met — seeds yes, dataset version no. Pinning the revision covers the dataset half.
